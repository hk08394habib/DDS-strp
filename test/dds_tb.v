module dds_tb();

   reg clk = 1'b0;
   reg reset = 1'b0;
   wire [7:0] phase_acc;

   wire [7:0] output_sin;

   phase_acc phase_acc_DUT(.clk(clk),
                           .mult(8'd16),
                           .reset(reset),
                           .phase_acc(phase_acc));

   sin_lookup sin_lookup_DUT(.clk(clk),
                             .address(phase_acc),
                             .value(output_sin));


   initial begin
      //$monitor("phase=%0d sin=%0d\n", phase_acc_DUT.phase_acc, sin_lookup_DUT.value);
      $display("start");
      clk = 1'b0;
      forever #1 clk = ~clk;
   end

   initial begin
      $dumpfile("counter.vcd");
      $dumpvars();
      #600
      $display("finish");
      $finish();
   end

endmodule;
