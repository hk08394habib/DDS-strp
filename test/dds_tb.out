#! /usr/bin/vvp
:ivl_version "12.0 (stable)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "/usr/lib64/ivl/system.vpi";
:vpi_module "/usr/lib64/ivl/vhdl_sys.vpi";
:vpi_module "/usr/lib64/ivl/vhdl_textio.vpi";
:vpi_module "/usr/lib64/ivl/v2005_math.vpi";
:vpi_module "/usr/lib64/ivl/va_math.vpi";
S_0x564ae535c510 .scope module, "dds_tb" "dds_tb" 2 1;
 .timescale 0 0;
v0x564ae53c5a00_0 .var "clk", 0 0;
v0x564ae53c5af0_0 .net "output_sin", 7 0, v0x564ae53c58a0_0;  1 drivers
v0x564ae53c5bb0_0 .net "phase_acc", 7 0, v0x564ae53c4e90_0;  1 drivers
v0x564ae53c5ca0_0 .var "reset", 0 0;
S_0x564ae535c6a0 .scope module, "phase_acc_DUT" "phase_acc" 2 9, 3 15 0, S_0x564ae535c510;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "clk";
    .port_info 1 /INPUT 8 "mult";
    .port_info 2 /INPUT 1 "reset";
    .port_info 3 /OUTPUT 8 "phase_acc";
P_0x564ae535c880 .param/l "f_clk" 0 3 22, +C4<00000000000011110100001001000000>;
v0x564ae5396e30_0 .net "clk", 0 0, v0x564ae53c5a00_0;  1 drivers
L_0x7fc0bf2b6018 .functor BUFT 1, C4<00010000>, C4<0>, C4<0>, C4<0>;
v0x564ae53c4db0_0 .net "mult", 7 0, L_0x7fc0bf2b6018;  1 drivers
v0x564ae53c4e90_0 .var "phase_acc", 7 0;
v0x564ae53c4f80_0 .net "reset", 0 0, v0x564ae53c5ca0_0;  1 drivers
E_0x564ae53a7740 .event negedge, v0x564ae53c4f80_0;
E_0x564ae53a8210 .event posedge, v0x564ae5396e30_0;
S_0x564ae53c50f0 .scope module, "sin_lookup_DUT" "sin_lookup" 2 14, 4 4 0, S_0x564ae535c510;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "clk";
    .port_info 1 /INPUT 8 "address";
    .port_info 2 /OUTPUT 8 "value";
P_0x564ae53c52f0 .param/l "CONST" 1 4 8, +C4<000000000000000000000000000111111>;
v0x564ae53c5420_0 .var "LUT_address", 5 0;
v0x564ae53c5520_0 .net "address", 7 0, v0x564ae53c4e90_0;  alias, 1 drivers
v0x564ae53c5610_0 .net "clk", 0 0, v0x564ae53c5a00_0;  alias, 1 drivers
v0x564ae53c5710_0 .var "shift_address", 7 0;
v0x564ae53c57b0_0 .var "sign", 0 0;
v0x564ae53c58a0_0 .var "value", 7 0;
E_0x564ae53a7ec0 .event anyedge, v0x564ae53c4e90_0;
    .scope S_0x564ae535c6a0;
T_0 ;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v0x564ae53c4e90_0, 0, 8;
    %end;
    .thread T_0;
    .scope S_0x564ae535c6a0;
T_1 ;
    %wait E_0x564ae53a8210;
    %load/vec4 v0x564ae53c4e90_0;
    %load/vec4 v0x564ae53c4db0_0;
    %add;
    %assign/vec4 v0x564ae53c4e90_0, 0;
    %load/vec4 v0x564ae53c4e90_0;
    %pad/u 32;
    %cmpi/ne 0, 0, 32;
    %flag_get/vec4 4;
    %jmp/0 T_1.2, 4;
    %load/vec4 v0x564ae53c4e90_0;
    %load/vec4 v0x564ae53c4db0_0;
    %cmp/u;
    %flag_get/vec4 5;
    %and;
T_1.2;
    %flag_set/vec4 8;
    %jmp/0xz  T_1.0, 8;
    %pushi/vec4 0, 0, 8;
    %assign/vec4 v0x564ae53c4e90_0, 0;
T_1.0 ;
    %load/vec4 v0x564ae53c4e90_0;
    %pad/u 32;
    %cmpi/e 0, 0, 32;
    %jmp/0xz  T_1.3, 4;
    %vpi_call 3 29 "$display", "starting" {0 0 0};
T_1.3 ;
    %jmp T_1;
    .thread T_1;
    .scope S_0x564ae535c6a0;
T_2 ;
    %wait E_0x564ae53a7740;
    %pushi/vec4 0, 0, 8;
    %assign/vec4 v0x564ae53c4e90_0, 0;
    %jmp T_2;
    .thread T_2;
    .scope S_0x564ae53c50f0;
T_3 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x564ae53c57b0_0, 0, 1;
    %end;
    .thread T_3;
    .scope S_0x564ae53c50f0;
T_4 ;
    %wait E_0x564ae53a7ec0;
    %vpi_call 4 21 "$display", "address is %0d", v0x564ae53c5520_0 {0 0 0};
    %load/vec4 v0x564ae53c5520_0;
    %pad/u 33;
    %cmpi/u 126, 0, 33;
    %flag_inv 5; GE is !LT
    %flag_get/vec4 5;
    %jmp/0 T_4.2, 5;
    %load/vec4 v0x564ae53c5520_0;
    %pad/u 33;
    %cmpi/u 252, 0, 33;
    %flag_get/vec4 4;
    %flag_get/vec4 5;
    %or;
    %and;
T_4.2;
    %flag_set/vec4 8;
    %jmp/0xz  T_4.0, 8;
    %load/vec4 v0x564ae53c5520_0;
    %pad/u 33;
    %subi 126, 0, 33;
    %pad/u 8;
    %store/vec4 v0x564ae53c5710_0, 0, 8;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x564ae53c57b0_0, 0, 1;
    %vpi_call 4 26 "$display", "address mapped to negative" {0 0 0};
    %jmp T_4.1;
T_4.0 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x564ae53c57b0_0, 0, 1;
    %load/vec4 v0x564ae53c5520_0;
    %store/vec4 v0x564ae53c5710_0, 0, 8;
    %vpi_call 4 30 "$display", "address mapped to positive" {0 0 0};
T_4.1 ;
    %vpi_call 4 33 "$display", "Shift address is %0d", v0x564ae53c5710_0 {0 0 0};
    %load/vec4 v0x564ae53c5710_0;
    %pad/u 32;
    %cmpi/u 0, 0, 32;
    %flag_inv 5; GE is !LT
    %flag_get/vec4 5;
    %jmp/0 T_4.5, 5;
    %load/vec4 v0x564ae53c5710_0;
    %pad/u 33;
    %cmpi/u 63, 0, 33;
    %flag_get/vec4 5;
    %and;
T_4.5;
    %flag_set/vec4 8;
    %jmp/0xz  T_4.3, 8;
    %load/vec4 v0x564ae53c5710_0;
    %pad/u 6;
    %store/vec4 v0x564ae53c5420_0, 0, 6;
    %vpi_call 4 38 "$display", "address mapped to first" {0 0 0};
    %jmp T_4.4;
T_4.3 ;
    %load/vec4 v0x564ae53c5710_0;
    %pad/u 33;
    %cmpi/u 63, 0, 33;
    %flag_inv 5; GE is !LT
    %flag_get/vec4 5;
    %jmp/0 T_4.8, 5;
    %load/vec4 v0x564ae53c5710_0;
    %pad/u 33;
    %cmpi/u 126, 0, 33;
    %flag_get/vec4 5;
    %and;
T_4.8;
    %flag_set/vec4 8;
    %jmp/0xz  T_4.6, 8;
    %pushi/vec4 126, 0, 33;
    %load/vec4 v0x564ae53c5710_0;
    %pad/u 33;
    %sub;
    %pad/u 6;
    %store/vec4 v0x564ae53c5420_0, 0, 6;
    %vpi_call 4 41 "$display", "address mapped to second" {0 0 0};
T_4.6 ;
T_4.4 ;
    %load/vec4 v0x564ae53c5420_0;
    %dup/vec4;
    %pushi/vec4 0, 0, 6;
    %cmp/u;
    %jmp/1 T_4.9, 6;
    %dup/vec4;
    %pushi/vec4 1, 0, 6;
    %cmp/u;
    %jmp/1 T_4.10, 6;
    %dup/vec4;
    %pushi/vec4 2, 0, 6;
    %cmp/u;
    %jmp/1 T_4.11, 6;
    %dup/vec4;
    %pushi/vec4 3, 0, 6;
    %cmp/u;
    %jmp/1 T_4.12, 6;
    %dup/vec4;
    %pushi/vec4 4, 0, 6;
    %cmp/u;
    %jmp/1 T_4.13, 6;
    %dup/vec4;
    %pushi/vec4 5, 0, 6;
    %cmp/u;
    %jmp/1 T_4.14, 6;
    %dup/vec4;
    %pushi/vec4 6, 0, 6;
    %cmp/u;
    %jmp/1 T_4.15, 6;
    %dup/vec4;
    %pushi/vec4 7, 0, 6;
    %cmp/u;
    %jmp/1 T_4.16, 6;
    %dup/vec4;
    %pushi/vec4 8, 0, 6;
    %cmp/u;
    %jmp/1 T_4.17, 6;
    %dup/vec4;
    %pushi/vec4 9, 0, 6;
    %cmp/u;
    %jmp/1 T_4.18, 6;
    %dup/vec4;
    %pushi/vec4 10, 0, 6;
    %cmp/u;
    %jmp/1 T_4.19, 6;
    %dup/vec4;
    %pushi/vec4 11, 0, 6;
    %cmp/u;
    %jmp/1 T_4.20, 6;
    %dup/vec4;
    %pushi/vec4 12, 0, 6;
    %cmp/u;
    %jmp/1 T_4.21, 6;
    %dup/vec4;
    %pushi/vec4 13, 0, 6;
    %cmp/u;
    %jmp/1 T_4.22, 6;
    %dup/vec4;
    %pushi/vec4 14, 0, 6;
    %cmp/u;
    %jmp/1 T_4.23, 6;
    %dup/vec4;
    %pushi/vec4 15, 0, 6;
    %cmp/u;
    %jmp/1 T_4.24, 6;
    %dup/vec4;
    %pushi/vec4 16, 0, 6;
    %cmp/u;
    %jmp/1 T_4.25, 6;
    %dup/vec4;
    %pushi/vec4 17, 0, 6;
    %cmp/u;
    %jmp/1 T_4.26, 6;
    %dup/vec4;
    %pushi/vec4 18, 0, 6;
    %cmp/u;
    %jmp/1 T_4.27, 6;
    %dup/vec4;
    %pushi/vec4 19, 0, 6;
    %cmp/u;
    %jmp/1 T_4.28, 6;
    %dup/vec4;
    %pushi/vec4 20, 0, 6;
    %cmp/u;
    %jmp/1 T_4.29, 6;
    %dup/vec4;
    %pushi/vec4 21, 0, 6;
    %cmp/u;
    %jmp/1 T_4.30, 6;
    %dup/vec4;
    %pushi/vec4 22, 0, 6;
    %cmp/u;
    %jmp/1 T_4.31, 6;
    %dup/vec4;
    %pushi/vec4 23, 0, 6;
    %cmp/u;
    %jmp/1 T_4.32, 6;
    %dup/vec4;
    %pushi/vec4 24, 0, 6;
    %cmp/u;
    %jmp/1 T_4.33, 6;
    %dup/vec4;
    %pushi/vec4 25, 0, 6;
    %cmp/u;
    %jmp/1 T_4.34, 6;
    %dup/vec4;
    %pushi/vec4 26, 0, 6;
    %cmp/u;
    %jmp/1 T_4.35, 6;
    %dup/vec4;
    %pushi/vec4 27, 0, 6;
    %cmp/u;
    %jmp/1 T_4.36, 6;
    %dup/vec4;
    %pushi/vec4 28, 0, 6;
    %cmp/u;
    %jmp/1 T_4.37, 6;
    %dup/vec4;
    %pushi/vec4 29, 0, 6;
    %cmp/u;
    %jmp/1 T_4.38, 6;
    %dup/vec4;
    %pushi/vec4 30, 0, 6;
    %cmp/u;
    %jmp/1 T_4.39, 6;
    %dup/vec4;
    %pushi/vec4 31, 0, 6;
    %cmp/u;
    %jmp/1 T_4.40, 6;
    %dup/vec4;
    %pushi/vec4 32, 0, 6;
    %cmp/u;
    %jmp/1 T_4.41, 6;
    %dup/vec4;
    %pushi/vec4 33, 0, 6;
    %cmp/u;
    %jmp/1 T_4.42, 6;
    %dup/vec4;
    %pushi/vec4 34, 0, 6;
    %cmp/u;
    %jmp/1 T_4.43, 6;
    %dup/vec4;
    %pushi/vec4 35, 0, 6;
    %cmp/u;
    %jmp/1 T_4.44, 6;
    %dup/vec4;
    %pushi/vec4 36, 0, 6;
    %cmp/u;
    %jmp/1 T_4.45, 6;
    %dup/vec4;
    %pushi/vec4 37, 0, 6;
    %cmp/u;
    %jmp/1 T_4.46, 6;
    %dup/vec4;
    %pushi/vec4 38, 0, 6;
    %cmp/u;
    %jmp/1 T_4.47, 6;
    %dup/vec4;
    %pushi/vec4 39, 0, 6;
    %cmp/u;
    %jmp/1 T_4.48, 6;
    %dup/vec4;
    %pushi/vec4 40, 0, 6;
    %cmp/u;
    %jmp/1 T_4.49, 6;
    %dup/vec4;
    %pushi/vec4 41, 0, 6;
    %cmp/u;
    %jmp/1 T_4.50, 6;
    %dup/vec4;
    %pushi/vec4 42, 0, 6;
    %cmp/u;
    %jmp/1 T_4.51, 6;
    %dup/vec4;
    %pushi/vec4 43, 0, 6;
    %cmp/u;
    %jmp/1 T_4.52, 6;
    %dup/vec4;
    %pushi/vec4 44, 0, 6;
    %cmp/u;
    %jmp/1 T_4.53, 6;
    %dup/vec4;
    %pushi/vec4 45, 0, 6;
    %cmp/u;
    %jmp/1 T_4.54, 6;
    %dup/vec4;
    %pushi/vec4 46, 0, 6;
    %cmp/u;
    %jmp/1 T_4.55, 6;
    %dup/vec4;
    %pushi/vec4 47, 0, 6;
    %cmp/u;
    %jmp/1 T_4.56, 6;
    %dup/vec4;
    %pushi/vec4 48, 0, 6;
    %cmp/u;
    %jmp/1 T_4.57, 6;
    %dup/vec4;
    %pushi/vec4 49, 0, 6;
    %cmp/u;
    %jmp/1 T_4.58, 6;
    %dup/vec4;
    %pushi/vec4 50, 0, 6;
    %cmp/u;
    %jmp/1 T_4.59, 6;
    %dup/vec4;
    %pushi/vec4 51, 0, 6;
    %cmp/u;
    %jmp/1 T_4.60, 6;
    %dup/vec4;
    %pushi/vec4 52, 0, 6;
    %cmp/u;
    %jmp/1 T_4.61, 6;
    %dup/vec4;
    %pushi/vec4 53, 0, 6;
    %cmp/u;
    %jmp/1 T_4.62, 6;
    %dup/vec4;
    %pushi/vec4 54, 0, 6;
    %cmp/u;
    %jmp/1 T_4.63, 6;
    %dup/vec4;
    %pushi/vec4 55, 0, 6;
    %cmp/u;
    %jmp/1 T_4.64, 6;
    %dup/vec4;
    %pushi/vec4 56, 0, 6;
    %cmp/u;
    %jmp/1 T_4.65, 6;
    %dup/vec4;
    %pushi/vec4 57, 0, 6;
    %cmp/u;
    %jmp/1 T_4.66, 6;
    %dup/vec4;
    %pushi/vec4 58, 0, 6;
    %cmp/u;
    %jmp/1 T_4.67, 6;
    %dup/vec4;
    %pushi/vec4 59, 0, 6;
    %cmp/u;
    %jmp/1 T_4.68, 6;
    %dup/vec4;
    %pushi/vec4 60, 0, 6;
    %cmp/u;
    %jmp/1 T_4.69, 6;
    %dup/vec4;
    %pushi/vec4 61, 0, 6;
    %cmp/u;
    %jmp/1 T_4.70, 6;
    %dup/vec4;
    %pushi/vec4 62, 0, 6;
    %cmp/u;
    %jmp/1 T_4.71, 6;
    %dup/vec4;
    %pushi/vec4 63, 0, 6;
    %cmp/u;
    %jmp/1 T_4.72, 6;
    %jmp T_4.73;
T_4.9 ;
    %pushi/vec4 100, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.10 ;
    %pushi/vec4 102, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.11 ;
    %pushi/vec4 104, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.12 ;
    %pushi/vec4 107, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.13 ;
    %pushi/vec4 109, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.14 ;
    %pushi/vec4 112, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.15 ;
    %pushi/vec4 114, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.16 ;
    %pushi/vec4 117, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.17 ;
    %pushi/vec4 119, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.18 ;
    %pushi/vec4 121, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.19 ;
    %pushi/vec4 124, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.20 ;
    %pushi/vec4 126, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.21 ;
    %pushi/vec4 129, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.22 ;
    %pushi/vec4 131, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.23 ;
    %pushi/vec4 133, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.24 ;
    %pushi/vec4 135, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.25 ;
    %pushi/vec4 138, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.26 ;
    %pushi/vec4 140, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.27 ;
    %pushi/vec4 142, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.28 ;
    %pushi/vec4 144, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.29 ;
    %pushi/vec4 147, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.30 ;
    %pushi/vec4 149, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.31 ;
    %pushi/vec4 151, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.32 ;
    %pushi/vec4 153, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.33 ;
    %pushi/vec4 155, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.34 ;
    %pushi/vec4 157, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.35 ;
    %pushi/vec4 159, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.36 ;
    %pushi/vec4 161, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.37 ;
    %pushi/vec4 163, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.38 ;
    %pushi/vec4 165, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.39 ;
    %pushi/vec4 167, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.40 ;
    %pushi/vec4 168, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.41 ;
    %pushi/vec4 170, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.42 ;
    %pushi/vec4 172, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.43 ;
    %pushi/vec4 174, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.44 ;
    %pushi/vec4 175, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.45 ;
    %pushi/vec4 177, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.46 ;
    %pushi/vec4 178, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.47 ;
    %pushi/vec4 180, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.48 ;
    %pushi/vec4 181, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.49 ;
    %pushi/vec4 183, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.50 ;
    %pushi/vec4 184, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.51 ;
    %pushi/vec4 185, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.52 ;
    %pushi/vec4 187, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.53 ;
    %pushi/vec4 188, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.54 ;
    %pushi/vec4 189, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.55 ;
    %pushi/vec4 190, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.56 ;
    %pushi/vec4 191, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.57 ;
    %pushi/vec4 192, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.58 ;
    %pushi/vec4 193, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.59 ;
    %pushi/vec4 194, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.60 ;
    %pushi/vec4 194, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.61 ;
    %pushi/vec4 195, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.62 ;
    %pushi/vec4 196, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.63 ;
    %pushi/vec4 197, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.64 ;
    %pushi/vec4 197, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.65 ;
    %pushi/vec4 198, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.66 ;
    %pushi/vec4 198, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.67 ;
    %pushi/vec4 198, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.68 ;
    %pushi/vec4 199, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.69 ;
    %pushi/vec4 199, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.70 ;
    %pushi/vec4 199, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.71 ;
    %pushi/vec4 199, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.72 ;
    %pushi/vec4 199, 0, 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
    %jmp T_4.73;
T_4.73 ;
    %pop/vec4 1;
    %load/vec4 v0x564ae53c57b0_0;
    %flag_set/vec4 8;
    %jmp/0xz  T_4.74, 8;
    %pushi/vec4 200, 0, 9;
    %load/vec4 v0x564ae53c58a0_0;
    %pad/u 9;
    %sub;
    %pad/u 8;
    %store/vec4 v0x564ae53c58a0_0, 0, 8;
T_4.74 ;
    %vpi_call 4 112 "$display", "%d\012", v0x564ae53c58a0_0 {0 0 0};
    %jmp T_4;
    .thread T_4, $push;
    .scope S_0x564ae535c510;
T_5 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x564ae53c5a00_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x564ae53c5ca0_0, 0, 1;
    %end;
    .thread T_5;
    .scope S_0x564ae535c510;
T_6 ;
    %vpi_call 2 21 "$display", "start" {0 0 0};
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x564ae53c5a00_0, 0, 1;
T_6.0 ;
    %delay 1, 0;
    %load/vec4 v0x564ae53c5a00_0;
    %inv;
    %store/vec4 v0x564ae53c5a00_0, 0, 1;
    %jmp T_6.0;
    %end;
    .thread T_6;
    .scope S_0x564ae535c510;
T_7 ;
    %vpi_call 2 27 "$dumpfile", "counter.vcd" {0 0 0};
    %vpi_call 2 28 "$dumpvars" {0 0 0};
    %delay 600, 0;
    %vpi_call 2 30 "$display", "finish" {0 0 0};
    %vpi_call 2 31 "$finish" {0 0 0};
    %end;
    .thread T_7;
# The file index is used to find the file name in the following table.
:file_names 5;
    "N/A";
    "<interactive>";
    "./test/dds_tb.v";
    "src/counter.v";
    "src/sin_lookup.v";
